Update the code review docs

johnpaulashenfelter opened the pull request:
Clean, reduce redundancy, and update.

Addresses #507

johnpaulashenfelter committed:
Update the code review docs

Clean, reduce redundancy, and update.

gunsch commented on platform/engineering/code_review_guidelines.md:
```diff
@@ -4,41 +4,58 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
+## The pull request review lifecycle in brief
+
+1. For initial review by your project team, create a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/)
+2. Once you are ready for the VSP team to review your PR, use the GitHub *Ready for review* button in the merge section of your pull request. An engineer should review the code within 2 days. 
```
this seems like a policy change, not just a guidelines/wording update (1 day --> 2 days), was that intentional?

johnpaulashenfelter replied:
1 business day like we discussed on slack

gunsch commented on platform/engineering/code_review_guidelines.md:
```diff
@@ -4,41 +4,58 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
+
+
+
+## Helpful hints
```
Can we pick a different title? This doc should be *requirements* we're imposing on teams sending things to us, not *advice*, and headings/wording should reflect that.

johnpaulashenfelter replied:
Done.

gunsch commented on platform/engineering/code_review_guidelines.md:
```diff
@@ -4,41 +4,58 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
-    * Before sending code to VSP team have it reviewed by your team (see **First review** below) using a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/).  
-      * This prevents [code owners](https://help.github.com/en/articles/about-code-owners) from being notified before pull request is ready for review.
-    * After having your pull request reviewed by your team you can click the *Ready for review* button in the merge section of your pull request.
-      * An engineer from the VSP team will aim to review within 24 hours (business days) after the pull request has been reviewed by your team and you have been marked it *Ready for review*. Each additional round of changes will be reviewed within 24 hours.  If you need a review faster, escalate via your DSVA product owner.
```
this update also loses the clarity about "additional rounds of changes", which IMO we should keep explicitly called out

johnpaulashenfelter replied:
Thanks! That literally just got dropped in edit.

johnpaulashenfelter committed:
Incorporating feedback

Added some suggestions from @gunsch

johnpaulashenfelter committed:
Incorporate more feedback

Added back the info about additional review rounds. Synced the language to use business days.

gunsch reviewed (approved):
LGTM, added @annaswims as a reviewer to take a look as well.

annaswims commented on platform/engineering/code_review_guidelines.md:
```diff
@@ -4,41 +4,55 @@ This document covers rules for engaging the VSP team in [pull request reviews](h
-
-* **Size/complexity**: keep your changes scoped to one feature at a time. This means:
-    * **Size**: changes should be smaller than **300** changed lines (not including tests or whitespace). *If your change can't be smaller than this, talk to the team in #vfs-platform-support before asking for a review.*
+We expect that most developers have worked with GitHub and review processes before, but want to be clear about our norms to reduce confusion and help code flow through the process as quickly and reliably as possible.
```
FWIW, Github is still relatively new to VA culture and git is not necessarily the dominant form of source control at VA https://dsva.slack.com/archives/C04KXPXL4/p1568909161000900

gunsch replied:
+1, but also it's still OK to state that as the _expectation_ for working with VSP

(maybe we can word away from "most  developers have worked with GitHub", but "we expect developers to be familiar with GitHub" is still true)

johnpaulashenfelter replied:
Thanks -- solid point to education me on 😄

johnpaulashenfelter committed:
Add Andrew's suggestion
